### DISCUSSION TITLE
Generate typings for component props

### DISCUSSION THREAD

**Discussion by bgourlie** [sveltejs/kit#5441] [Ideas]
It would be great if, in addition to `RequestHandler` and `Load`, a `Props` type was generated. My (somewhat naive) assumption is that it should be reasonably straightforward to parse out types for all `export let`s defined in a component. One issue that comes to mind would be correctly importing any custom types.

**Comment by benmccann** [comment]
See https://github.com/sveltejs/kit/issues/647
